src/pages/Proposal.vue

```diff
@@ -144,9 +151,116 @@ export default defineComponent({
 			ProposalType,
 		};
 	},
+	setup() {
+		const formState = reactive(ProposalStore());
+		const walletStore = reactive(WalletStore());
+		const daoStore = reactive(DaoID());
+
+		return {
+			formState,
+			walletStore,
+			daoStore,
+			validateMessages: VALIDATE_MESSAGES,
+		};
+	},
 	methods: {
-		onFinish(values: Event) {
-			console.log("Success:", values);
+		async onFinish(values: any) {
+			try {
+				let {
+					amount,
+					recipient,
+					url,
+					url_hash,
+					proposal_type,
+					vote_date,
+					message,
+					asaId,
+				} = values;
+				if (typeof this.daoStore.dao_id === "undefined") {
+					console.error("appId not defined");
+					return;
+				}
+				if (typeof this.daoStore.govt_id === "undefined") {
+					console.error("govt_id not defined");
+					return;
+				}
```

Comment:
we should display the error to UI, along with console, so that user knows what went wrong

Reply:
Done